contracts/core/Factory.sol

```diff
@@ -51,7 +52,12 @@ contract Factory is IFactory {
       token1 = token0;
       token0 = WETH;
     }
+
+    // what's the reason for not providing all these parameters in the constructor?
+    // But at the same time, the pattern of assigning a global var, read then delete is interesting
+    // It's the first time I see this pattern
```

Comment:
This was originally found in UniswapV3 and I thought it was interesting to use. As Lam said, we can't use a variables in the constructor if not the init code/creation code of Pool will change everytime since we have different argument values when constructing the Pool contract

The Params is being reused every single time a Pool is being deployed. This is different from initialize where the factory needs to deploy then make an external call to initialize. The reason for deleting I believe is because of the refund in gas (from STORAGEKILL? which is -15k since to SSTORE it'll take 20k)

Reply:
- Oh I don't know that this pattern is in UniV3, interesting
- Init Code issue has been explained in the server, please check
- Any benefits for the factory to not initialize but instead the pool calls the factory? I thought it's kinda the same?
- Yeah deleting is a very good thing actually